application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -123,4 +135,47 @@ private RestAction<Message> sendHelperHeadsUp(ThreadChannel threadChannel) {
         return threadChannel.sendMessage(headsUpWithoutRole)
             .flatMap(message -> message.editMessage(headsUpWithRole));
     }
+
+    @Override
+    public String getName() {
+        return "chatpgt-answer";
+    }
+
+    @Override
+    public UserInteractionType getInteractionType() {
+        return UserInteractionType.OTHER;
+    }
+
+    @Override
+    public void acceptComponentIdGenerator(ComponentIdGenerator generator) {
+        componentIdInteractor.acceptComponentIdGenerator(generator);
+    }
+
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        ThreadChannel channel = event.getChannel().asThreadChannel();
+        Member interactionUser = event.getMember();
+        if (channel.getOwnerIdLong() != interactionUser.getIdLong()
```

Comment:
please add a comment to make it easier to understand which button this is all about. this class is not a chatgpt class after all, but the general help thread thing.

tbh, im really not a fan of how the chatgpt thing is mixed into the helpthread class, instead of having some sort of extra class (not ur fault though)

Reply:
I'm also not a fan of this, would it be possible to split it into a different class?

Reply:
sure. but its totally okay if you dont want to do this in this PR. just make sure to add comments instead then

Reply:
tbh I would encapsulate all the ChatGPT stuff out of the HelpSystem classes, it doesn't really fit in there ngl
but if moving all of it, it makes more sense to make a separate pr for that